infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/PVCSubPathHelper.java

```diff
@@ -218,16 +227,21 @@ private void execute(
     KubernetesDeployments deployments = null;
     try {
       deployments = factory.access(workspaceId, namespace).deployments();
-      deployments.create(pod);
-      watchLogsIfDebugEnabled(deployments, pod, identity, startOptions);
-      final Pod finished = deployments.wait(podName, WAIT_POD_TIMEOUT_MIN, POD_PREDICATE::apply);
-      PodStatus finishedStatus = finished.getStatus();
-      if (POD_PHASE_FAILED.equals(finishedStatus.getPhase())) {
-        String logs = deployments.getPodLogs(podName);
-        LOG.error(
-            "Job command '{}' execution is failed. Logs: {}",
-            Arrays.toString(command),
-            Strings.nullToEmpty(logs).replace("\n", " \\n")); // Force logs onto one line
+
+      if (pvcExistsAndBound(namespace, pvcName)) {
```

Comment:
anyway I would not increase code nesting and go in the following way:
```java
if (!doesPvcExist(pvc)) {
    // no PVC exist, so nothing to clean up
	return;
}

// do clean up
```
and yeah, this check could be done much earlier even before pod object creation, or even on other more high level class.

Reply:
not only should we check that PVC does not exist, but also that it is not in a terminating state.